src/vanilla/utils/atomWithStorage.ts

```diff
@@ -99,8 +99,18 @@ export function createJSONStorage<Value>(
   getStringStorage: () =>
     | AsyncStringStorage
     | SyncStringStorage
-    | undefined = () =>
-    typeof window !== 'undefined' ? window.localStorage : undefined,
+    | undefined = () => {
+    try {
+      const storage =
+        typeof window !== 'undefined' ? window.localStorage : undefined
+      return storage
```

Comment:
Can we do this and catch "undefined" error below?
```suggestion
      return window.localStorage
```